x-pack/metricbeat/cmd/root.go

```diff
@@ -5,16 +5,37 @@
 package cmd
 
 import (
-	"github.com/elastic/beats/metricbeat/cmd"
+	"flag"
+
+	"github.com/spf13/pflag"
+
+	cmd "github.com/elastic/beats/libbeat/cmd"
+	"github.com/elastic/beats/libbeat/cmd/instance"
+	mbcmd "github.com/elastic/beats/metricbeat/cmd"
+	"github.com/elastic/beats/metricbeat/cmd/test"
 	xpackcmd "github.com/elastic/beats/x-pack/libbeat/cmd"
+	"github.com/elastic/beats/x-pack/metricbeat/beater"
+	xpacktest "github.com/elastic/beats/x-pack/metricbeat/cmd/test"
 
 	// Register the includes.
 	_ "github.com/elastic/beats/x-pack/metricbeat/include"
+
+	// Import OSS modules.
+	_ "github.com/elastic/beats/metricbeat/include"
+	_ "github.com/elastic/beats/metricbeat/include/fields"
 )
 
+// Name of this beat
+var Name = "metricbeat"
+
 // RootCmd to handle beats cli
-var RootCmd = cmd.RootCmd
+var RootCmd *cmd.BeatsRootCmd
 
 func init() {
-	xpackcmd.AddXPack(RootCmd, cmd.Name)
+	var runFlags = pflag.NewFlagSet(Name, pflag.ExitOnError)
+	runFlags.AddGoFlag(flag.CommandLine.Lookup("system.hostfs"))
+	RootCmd = cmd.GenRootCmdWithSettings(beater.Creator(), instance.Settings{RunFlags: runFlags, Name: Name})
```

Comment:
maybe we should construct the Creator here instead of using beater.Creator. Reduces the chance of goimports to accidentily importing the wrong package in the future.

Reply:
I will give a try to create the creators where they are used.